src/transformers/models/llama/modeling_llama.py

```diff
@@ -1051,14 +1028,8 @@ def _update_causal_mask(self, attention_mask, input_tensor):
             )
 
         if self.config._attn_implementation == "sdpa":
-            if attention_mask is None:
-                return None
-            is_tracing = torch.jit.is_tracing() or isinstance(input_tensor, torch.fx.Proxy)
-            if not is_tracing and (torch.all(attention_mask == 1)):
-                return None
-            if is_tracing and seq_length == 1:
-                return None
```

Comment:
all of this failed generations, deal with it later

Reply:
cc @fxmarty I am warning you in advance  🥶 you might have to do something similar to the `prepared_4d_sdpa` but this is a lot simpler so for the better